colcon_core/shell/template/prefix_util.py.em

```diff
@@ -311,6 +316,20 @@ def _prepend_unique_value(name, value):
     return [line]
 
 
+# generate commands for removing prepended underscores
+def _remove_trailing_separators():
+    # do nothing if the shell extension does not implement the logic
+    if FORMAT_STR_REMOVE_TRAILING_SEPARATOR is None:
+        return []
+
+    global env_state
+    commands = []
+    for name in env_state:
+        commands += [FORMAT_STR_REMOVE_TRAILING_SEPARATOR.format_map(
```

Comment:
Should this be limited to to the ones which this logic added the trailing separator? That should be the ones which are not set / empty in `os.environ`.

Reply:
I guess that wouldn't hurt and is more conservative.

Reply:
Done dfca941